lib/web_ui/lib/src/engine/surface/surface.dart

```diff
@@ -483,6 +483,39 @@ enum PersistedSurfaceState {
   released,
 }
 
+class PersistedSurfaceException implements Exception {
+  PersistedSurfaceException(this.surface, this.message);
+
+  final PersistedSurface surface;
+  final String message;
+
+  @override
+  String toString() {
+    if (assertionsEnabled) {
+      return '${surface.runtimeType}: $message';
+    }
+    return super.toString();
+  }
+}
+
+/// Verifies that the [surface] is in one of the valid states.
+///
+/// This function should be used inside an assertion expression.
+bool debugAssertSurfaceState(PersistedSurface surface, PersistedSurfaceState state1, [PersistedSurfaceState state2, PersistedSurfaceState state3]) {
```

Comment:
I'm curious why you chose this signature vs passing an array of states.

Reply:
Which one is better

This?

```dart
assert(debugAssertSurfaceState(oldSurface, PersistedSurfaceState.active, PersistedSurfaceState.pendingUpdate));
```

Or this?

```dart
assert(debugAssertSurfaceState(oldSurface, const <PersistedSurfaceState>[PersistedSurfaceState.active, PersistedSurfaceState.pendingUpdate]))
```

Reply:
Ah right. Explicit types :(